cgroups/fs/apply_raw.go

```diff
@@ -118,21 +137,25 @@ func GetStats(systemPaths map[string]string) (*cgroups.Stats, error) {
 
 // Freeze toggles the container's freezer cgroup depending on the state
 // provided
-func Freeze(c *cgroups.Cgroup, state cgroups.FreezerState) error {
-	d, err := getCgroupData(c, 0)
+func (m *Manager) Freeze(state cgroups.FreezerState) error {
+	d, err := getCgroupData(m.Cgroups, 0)
 	if err != nil {
 		return err
 	}
 
-	c.Freezer = state
-
 	freezer := subsystems["freezer"]
```

Comment:
can we check here?

freeze, ok := subsystems["freezer"]
if !ok {
  return fmt.Errorf("the freezer subsystem is not present in this container")
}